clients/cpp-console/include/AudioPlayer.h

```diff
@@ -99,31 +109,79 @@ class IAudioPlayer{
         /// <example>
         /// <code>
         /// IAudioPlayer *audioPlayer = new LinuxAudioPlayer();
-        /// audioPlayer->Open();
+        /// audioPlayer->Initialize();
         /// ... 
         ///
         /// //In the Activity received callback
         /// if (event.HasAudio()){
         ///     std::shared_ptr<Audio::PullAudioOutputStream> stream = event.GetAudio();
-        ///     audioPLayer->Play(stream);
+        ///     int result = audioPLayer->Play(stream);
+        ///     if(result < 0){
+        ///         //error
+        ///     }
         /// }
         /// </code>
         /// </example>
         /// <remarks>
         /// Here we use the LinuxAudioPlayer as an example. This is preferred to the Byte array if possible
         /// since this will not cause copies of the buffer to be stored at runtime.
-        /// In our implementation we assume Open is called before playing.
+        /// In our implementation we assume Initialize is called before playing.
         /// </remarks>
         virtual int Play(std::shared_ptr<Microsoft::CognitiveServices::Speech::Audio::PullAudioOutputStream> pStream) = 0;
```

Comment:
Can you add comments regarding how playback automatically stops? what do I need to do with the stream to mark it "complete" (write a zero-size butter into it)

Reply:
On the same note, in the other play method (virtual int Play(uint8_t* buffer, size_t bufferSize)), how do I tell it when I'm done with my audio data? how about saying that if you call the method with bufferSize = 0, the player knows not to expect any more input buffers and it will stop playback when it's done playing from the queue. The state will change back to Initialized.

Reply:
the calls to play are complete. Once called, you are giving control over to the player. The caller does not control the audio data any more and the player is the one that stops in whatever way it needs to.

Reply:
Let's have a Teams call about this.

Reply:
Per our phone call, we will refine the policy of when we initialize and stop the player in a separate check in. Thanks!